[clickhouse] can't show progress for some query

**Summary**

If we run this query in ClickHouse client:
```
SELECT number FROM numbers_mt(10000000000) ORDER BY number DESC LIMIT 100;
```
It can not show the progress, but the group by query is ok:
```
datafuse :) SELECT max(number), sum(number) FROM numbers_mt(1000000000) GROUP BY number % 3, number % 4, number % 5;
```

It seems that only the group by query working.

Yes, that's meet my test to



> 
> ```
> RUSTFLAGS="-C target-cpu=native" cargo build --bin=fuse-query --features allocator --release
> ```

That's not work on my environment:

```
.cargo/registry/src/github.com-1ecc6299db9ec823/snmalloc-sys-0.2.27/snmalloc/src/override/../mem/../mem/../pal/pal_posix.h:294:23: error: ‘getentropy’ was not declared in this scope; did you mean ‘get_entropy64’?
    294 |         if (getentropy(&result, sizeof(result)) != 0)
        |             ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
        |             get_entropy64
  make[3]: *** [CMakeFiles/snmallocshim-1mib-rust.dir/src/override/rust.cc.o] Error 1
  make[2]: *** [CMakeFiles/snmallocshim-1mib-rust.dir/all] Error 2
  make[1]: *** [CMakeFiles/snmallocshim-1mib-rust.dir/rule] Error 2
  make: *** [snmallocshim-1mib-rust] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

BTW, where I can find the newest progress of this project? Currently I cannot create a local table and do some insert & query test right ?

--- a/query/src/servers/clickhouse/interactive_worker_base.rs
+++ b/query/src/servers/clickhouse/interactive_worker_base.rs
@@ -27,6 +27,7 @@ use common_clickhouse_srv::types::Block as ClickHouseBlock;
 use common_clickhouse_srv::CHContext;
 use common_datablocks::DataBlock;
 use common_datavalues::DataSchemaRef;
+use common_exception::ErrorCode;
 use common_exception::Result;
 use common_planners::InsertPlan;
 use common_planners::PlanNode;
@@ -72,48 +73,55 @@ impl InteractiveWorkerBase {
             _ => {
                 let start = Instant::now();
                 let interpreter = InterpreterFactory::get(ctx.clone(), plan)?;
-                // Write start query log.
-                let _ = interpreter
-                    .start()
-                    .await
-                    .map_err(|e| tracing::error!("interpreter.start.error: {:?}", e));
-
                 let name = interpreter.name().to_string();
-                let async_data_stream = interpreter.execute(None);
-                let mut data_stream = async_data_stream.await?;
                 histogram!(
                     super::clickhouse_metrics::METRIC_INTERPRETER_USEDTIME,
                     start.elapsed(),
                     "interpreter" => name
                 );
-                let mut interval_stream = IntervalStream::new(interval(Duration::from_millis(30)));
-                let cancel = Arc::new(AtomicBool::new(false));
 
-                let (mut tx, rx) = mpsc::channel(20);
-                let mut tx2 = tx.clone();
+                let cancel = Arc::new(AtomicBool::new(false));
                 let cancel_clone = cancel.clone();
 
+                let (tx, rx) = mpsc::channel(20);
+                let mut data_tx = tx.clone();
+                let mut progress_tx = tx;
+
                 let progress_ctx = ctx.clone();
+                let mut interval_stream = IntervalStream::new(interval(Duration::from_millis(30)));
                 tokio::spawn(async move {
                     while !cancel.load(Ordering::Relaxed) {
                         let _ = interval_stream.next().await;
                         let values = progress_ctx.get_and_reset_scan_progress_value();
-                        tx.send(BlockItem::ProgressTicker(values)).await.ok();
+                        progress_tx
+                            .send(BlockItem::ProgressTicker(values))
+                            .await
+                            .ok();
                     }
                 });
 
                 ctx.try_spawn(async move {
-                    while let Some(block) = data_stream.next().await {
-                        tx2.send(BlockItem::Block(block)).await.ok();
-                    }
+                    // Query log start.
+                    let _ = interpreter
+                        .start()
+                        .await
+                        .map_err(|e| tracing::error!("interpreter.start.error: {:?}", e));
 
+                    // Execute and read stream data.
+                    let async_data_stream = interpreter.execute(None);
+                    let mut data_stream = async_data_stream.await?;
+                    while let Some(block) = data_stream.next().await {
+                        data_tx.send(BlockItem::Block(block)).await.ok();
+                    }
                     cancel_clone.store(true, Ordering::Relaxed);
+
+                    // Query log finish.
+                    let _ = interpreter
+                        .finish()
+                        .await
+                        .map_err(|e| tracing::error!("interpreter.finish.error: {:?}", e));
+                    Ok::<(), ErrorCode>(())
                 })?;
-                // Write final query log.
-                let _ = interpreter
-                    .finish()
-                    .await
-                    .map_err(|e| tracing::error!("interpreter.finish.error: {:?}", e));
 
                 Ok(rx)
             }

